app/models/course/video.rb

```diff
@@ -99,7 +105,8 @@ def initialize_duplicate_tab(duplicator, other)
   end
 
   def url_unchanged
-    errors.add(:url, 'should not be updated for existing videos') if url_changed? &&
-                                                                     persisted?
+    errors.add(:url, 'cannot be updated for videos with comments or watch data') if url_changed? &&
+                                                                                    persisted? &&
```

Comment:
I think you can shift this to the validate declaration: `validate :url_unchanged on: :update`.